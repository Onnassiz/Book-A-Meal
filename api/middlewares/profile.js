import isUrl from 'is-url';

const { check } = require('express-validator/check');

export const profileFormConstraints = [
  check('businessName')
    .exists()
    .withMessage('the business name field is require')
    .isLength({ min: 1 })
    .withMessage('the business name field is required')
    .isString()
    .withMessage('the business name must be a string')
    .trim(),

  check('mission')
    .exists()
    .withMessage('the business name field is require')
    .isLength({ min: 1 })
    .withMessage('the business name field is required')
    .isString()
    .withMessage('the business name must be a string')
    .trim(),

  check('contact')
    .exists()
    .withMessage('the contact field is require')
    .isLength({ min: 1 })
    .withMessage('the contact field is required')
    .isString()
    .withMessage('the contact must must be a string')
    .trim(),

  check('email')
    .exists()
    .withMessage('the email field is require')
    .isLength({ min: 1 })
    .withMessage('the email field is required')
    .isEmail()
    .withMessage('the email field must be an email')
    .trim()
    .normalizeEmail(),

  check('banner')
    .optional({ nullable: true })
    .custom((value) => {
      return isUrl(value);
    })
    .withMessage('image link must be a URL'),
];

export const profileBannerConstraints = [
  check('banner')
    .exists()
    .withMessage('the image field is require')
    .isLength({ min: 1 })
    .withMessage('the image field is required')
    .optional({ nullable: true })
    .custom((value) => {
      return isUrl(value);
    })
    .withMessage('image link must be a URL'),
];
