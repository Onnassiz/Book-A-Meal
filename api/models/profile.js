
module.exports = (sequelize, DataTypes) => {
  const Profile = sequelize.define('profile', {
    id: {
      allowNull: false,
      primaryKey: true,
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
    },
    businessName: DataTypes.STRING,
    contact: DataTypes.STRING,
    email: DataTypes.STRING,
    mission: DataTypes.STRING,
    banner: DataTypes.STRING,
  }, {});
  Profile.associate = (models) => {
    Profile.belongsTo(models.user);
    // Profile.belongsTo(models.mealOrder);
  };
  return Profile;
};

