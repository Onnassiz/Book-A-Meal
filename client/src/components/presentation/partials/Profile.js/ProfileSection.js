import React from 'react';
import PropTypes from 'prop-types';
import empty from 'is-empty';

const ProfileSection = props => (
  <div className="col-12" >
    <div id="detail">
      <h2>Business Name</h2><span>{props.profile.businessName}</span>
    </div>
    <div id="detail">
      <h2>Mission</h2>
      <span>{props.profile.mission}</span>
    </div>
    <div id="detail">
      <h2>Business Email</h2>
      <span>{props.profile.email}</span>
    </div>
    <div id="detail">
      <h2>Contact</h2>
      <span>{props.profile.contact}</span>
    </div>
    <div id="detail">
      <h2>Banner</h2>
      <span>
        {empty(props.profile.banner) ? <i className="material-icons">photo</i> :
        <img src={props.profile.banner} alt="banner" style={{ maxHeight: '600px', borderRadius: 4 }} />}
      </span>
    </div>
  </div>
);


ProfileSection.propTypes = {
  profile: PropTypes.object.isRequired,
};

export default ProfileSection;

