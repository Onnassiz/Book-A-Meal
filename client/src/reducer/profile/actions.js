import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actions';

export const SET_PROFILE = 'SET_PROFILE';
export const SET_PROFILE_SERVER_ERRORS = 'SET_PROFILE_SERVER_ERRORS';
export const SET_PROFILE_ALERT = 'SET_PROFILE_ALERT';

function setAlert(alert) {
  return dispatch => dispatch({
    type: SET_PROFILE_ALERT,
    alert,
  });
}

export function setProfile(profile) {
  return dispatch => dispatch({
    type: SET_PROFILE,
    id: profile.id,
    businessName: profile.businessName,
    mission: profile.mission,
    contact: profile.contact,
    banner: profile.banner,
    email: profile.email,
  });
}

function setProfileErrors(errors) {
  return dispatch => dispatch({
    type: SET_PROFILE_SERVER_ERRORS,
    errors,
  });
}

export function postProfile(profile) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.post('api/v1/profile', profile).then((response) => {
      dispatch(unsetLoading());
      dispatch(setProfile(response.data));
      dispatch(setAlert('Profile successfully created'));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setProfileErrors(error.response.data));
      return error;
    });
  };
}

export function updateProfile(profile) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.put(`api/v1/profile/${profile.id}`, profile).then((response) => {
      dispatch(unsetLoading());
      dispatch(setProfile(response.data));
      dispatch(setAlert('Profile successfully updated'));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setProfileErrors(error.response.data));
      return error;
    });
  };
}

export function putImage(id, field) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.put(`api/v1/profile/image/${id}`, field).then((response) => {
      dispatch(unsetLoading());
      dispatch(setProfile(response.data));
      dispatch(setAlert('Banner successfully uploaded'));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setProfileErrors(error.response.data));
      return error;
    });
  };
}

export function getProfile() {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.get('api/v1/profile').then((response) => {
      dispatch(unsetLoading());
      dispatch(setProfile(response.data));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      return error;
    });
  };
}
