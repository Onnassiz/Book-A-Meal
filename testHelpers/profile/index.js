import { profile } from '../../api/models';

import { getCatererId } from '../main';

const deleteProfile = (done) => {
  profile.destroy({
    where: {},
  }).then(() => {
    done();
  });
};

const insertProfile = done => getCatererId(done, false).then((id) => {
  const newProfile = profile.build({
    businessName: 'Your name',
    mission: 'Your mission',
    contact: 'Your this is your contact',
    banner: 'https://banner.com',
    email: 'benjamin@gmail.com',
    userId: id,
  });

  return newProfile.save().then(data => data);
});

export {
  deleteProfile,
  insertProfile,
};
