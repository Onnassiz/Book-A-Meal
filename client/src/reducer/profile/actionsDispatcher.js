import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actionsDispatcher';
import { setProfileAction, setProfileErrorsAction } from './actions';


export const setProfile = profile => dispatch => dispatch(setProfileAction(profile));

const setProfileErrors = errors => dispatch => dispatch(setProfileErrorsAction(errors));

export const postProfile = profile => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/profile', profile).then((response) => {
    dispatch(unsetLoading());
    dispatch(setProfile(response.data));
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setProfileErrors(error.response.data));
    return error;
  });
};

export const updateProfile = profile => (dispatch) => {
  dispatch(setLoading());
  return axios.put(`api/v1/profile/${profile.id}`, profile).then((response) => {
    dispatch(unsetLoading());
    dispatch(setProfile(response.data));
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setProfileErrors(error.response.data));
    return error;
  });
};

export const putImage = (id, field) => (dispatch) => {
  dispatch(setLoading());
  return axios.put(`api/v1/profile/${id}`, field).then((response) => {
    dispatch(unsetLoading());
    dispatch(setProfile(response.data));
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setProfileErrors(error.response.data));
    return error;
  });
};

export const getProfile = () => (dispatch) => {
  dispatch(setLoading());
  return axios.get('api/v1/profile').then((response) => {
    dispatch(unsetLoading());
    dispatch(setProfile(response.data));
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setProfileErrors(error.response.data));
    return error;
  });
};
