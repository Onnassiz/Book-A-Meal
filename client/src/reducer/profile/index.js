import { SET_PROFILE, SET_PROFILE_SERVER_ERRORS } from './actions';

const initialState = {};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_PROFILE:
      return {
        id: action.id,
        businessName: action.businessName,
        email: action.email,
        mission: action.mission,
        contact: action.contact,
        banner: action.banner,
      };
    case SET_PROFILE_SERVER_ERRORS:
      return { ...state, errors: action.errors };
    default:
      return state;
  }
};
