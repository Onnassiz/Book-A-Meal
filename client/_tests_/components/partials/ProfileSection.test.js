import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProfileSection from '../../../src/components/presentation/partials/Profile/ProfileSection';
import profile from '../../objectProps/profile.props';

const props = {
  profile,
};

const wrapper = shallow(<ProfileSection {...props} />);

describe('<ProfileSection />', () => {
  it('Should render ProfileSection  component successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
