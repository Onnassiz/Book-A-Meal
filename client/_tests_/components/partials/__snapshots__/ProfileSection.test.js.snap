// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileSection /> Should render ProfileSection  component successfully 1`] = `
<div
  className="col-12"
>
  <div
    id="detail"
  >
    <h2>
      Business Name
    </h2>
    <span>
      The blast
    </span>
  </div>
  <div
    id="detail"
  >
    <h2>
      Mission
    </h2>
    <span>
      Feeding the world
    </span>
  </div>
  <div
    id="detail"
  >
    <h2>
      Business Email
    </h2>
    <span>
      onnassiz@andela.com
    </span>
  </div>
  <div
    id="detail"
  >
    <h2>
      Contact
    </h2>
    <span>
      12 Benny Road Lagos
    </span>
  </div>
  <div
    id="detail"
  >
    <h2>
      Banner
    </h2>
    <span>
      <img
        alt="banner"
        src="See me coming"
        style={
          Object {
            "borderRadius": 4,
            "maxHeight": "600px",
          }
        }
      />
    </span>
  </div>
</div>
`;
